contracts/upgradable/Upgradable.sol

```diff
@@ -50,6 +50,11 @@ abstract contract Upgradable is Ownable, Implementation, IUpgradable {
             revert InvalidImplementation();
 
         if (newImplementationCodeHash != newImplementation.codehash) revert InvalidCodeHash();
+        if (newImplementation == implementation()) revert InvalidImplementation();
```

Comment:
Why do we need this restriction? It can be useful to upgrade to the same implementation for the purpose of calling setup (say to migrate something, such as operators etc.)